### DISCUSSION TITLE
Endpoints and collections management on applications deletion

### DISCUSSION THREAD

**Discussion by graz-dev** [mia-platform/community#400] [General]
### Description

Hello, I'd like to report an issue I encountered while experimenting with the Dev Portal application. When I install the application, it includes the CRUD service along with several collections and an API gateway with a series of endpoints. The problem arises when I try to uninstall the application: all microservices are correctly removed, but the endpoints and collections remain configured. This becomes problematic because I cannot manually delete these through the console UI since the related services are no longer installed. I'm forced to do this manually through the configurations in the repository.

**Comment by Danielecina** [comment]
Hi @graz-dev, What do you mean by uninstalling an application?
Do you intend to delete an application from the project?

In that case when you delete an application, I would expect the services to remain in the list of services in the console and then later one by one you go and delete the individual microservices. 

Deleting an individual microservices involves cleaning up endpoints and collection connected to it
What version of the console are you using?

**Reply by fredmaggiowski** [reply]
>Deleting an individual microservices involves cleaning up endpoints and collection connected to it

this is not true.


The console will not delete any endpoint associated with an API Gateway or any colletion associated with the crud-service.

The reason for this is to make the user capable of switching the service (particularly useful when migrating from the Nginx API Gateway to the Envoy-based API Gateway) without the need to redefine all the endpoints.

**Comment by graz-dev** [comment]
Yep, I test it again and it was my bad, sorry!
